addons/html_editor/static/src/main/table/table_drag_drop.js

```diff
@@ -0,0 +1,184 @@
+import { Component, useRef, useExternalListener } from "@odoo/owl";
+import { BORDER_SENSITIVITY } from "./table_plugin";
+
+export class TableDragDrop extends Component {
+    static template = "html_editor.TableDragDrop";
+    static props = {
+        overlayRect: Object,
+        pointerPos: Object,
+        type: String,
+        close: Function,
+        turnIntoRow: { type: Function, optional: true },
+        turnIntoHeader: { type: Function, optional: true },
+        addStep: Function,
+        target: { validate: (el) => el.nodeType === Node.ELEMENT_NODE },
+        document: { validate: (p) => p.nodeType === Node.DOCUMENT_NODE },
+    };
+
+    setup() {
+        this.overlayRef = useRef("dragOverlay");
+        this.overlayRect = { ...this.props.overlayRect };
+        this.pointerPos = { ...this.props.pointerPos };
+        this.tableElement = closestElement(this.props.target, "table");
+        this.tableBounds = this.tableElement.getBoundingClientRect();
+        // Compute bounding rects of rows or column cells
+        this.itemRects =
+            this.props.type === "row"
+                ? [...this.tableElement.rows].map((r) => r.getBoundingClientRect())
+                : [...this.props.target.parentElement.cells].map((c) => c.getBoundingClientRect());
+
+        useExternalListener(this.props.document, "pointermove", this.onPointerMove);
+        useExternalListener(this.props.document, "pointerup", this.onPointerUp);
+        if (this.props.document !== document) {
+            // Listen outside the iframe.
+            useExternalListener(document, "pointermove", this.onPointerMove);
+            useExternalListener(document, "pointerup", this.onPointerUp);
+        }
+    }
+
+    clearHighlights() {
+        // Determine which highlight classes to remove.
+        const highlightClasses =
+            this.props.type === "row"
+                ? ["tr-highlight-top", "tr-highlight-bottom"]
+                : ["td-highlight-left", "td-highlight-right"];
+        const highlightedElements = this.tableElement.querySelectorAll(
+            highlightClasses.map((cls) => `.${cls}`).join(", ")
+        );
+        // Remove the highlight classes.
+        highlightedElements.forEach((el) => el.classList.remove(...highlightClasses));
+    }
+
+    getInsertPosition() {
+        // Calculate middle of overlay(vertical for row, horizontal for column)
+        const overlayMiddle =
+            this.props.type === "row"
+                ? this.overlayRect.top + this.overlayRect.height / 2
+                : this.overlayRect.left + this.overlayRect.width / 2;
+        // Index of target row/column where overlay is currently positioned
+        const targetIndex = this.itemRects.findIndex((rect) =>
+            this.props.type === "row"
+                ? rect.top <= overlayMiddle && overlayMiddle <= rect.bottom
+                : rect.left <= overlayMiddle && overlayMiddle <= rect.right
```

Comment:
```suggestion
            this.props.type === "row"
                ? overlayMiddle >= rect.top  && overlayMiddle <= rect.bottom
                : overlayMiddle >= rect.left && overlayMiddle <= rect.right
```
Same thing but I feel this is simpler to understand. :)